python/sglang/srt/managers/schedule_policy.py

```diff
@@ -442,46 +439,40 @@ def add_req_state(r, insert_sort=False):
 
         return self.budget_state()
 
-    def add_one_req(
-        self, req: Req, has_chunked_req: bool, enable_hierarchical_cache: bool = False
-    ):
+    def add_one_req(self, req: Req, has_chunked_req: bool):
         if req.sampling_params.ignore_eos and getattr(self.tree_cache, "disable", True):
             return self.add_one_req_ignore_eos(req, has_chunked_req)
 
         total_tokens = req.extend_input_len + min(
             req.sampling_params.max_new_tokens, CLIP_MAX_NEW_TOKENS_ESTIMATION
         )
-        input_tokens = (
-            -(-req.extend_input_len // self.tree_cache.page_size)
-            * self.tree_cache.page_size
-        )
+
+        # we should consider the possible host indices length when scheduling
+        min_extend_tokens = req.extend_input_len - req.host_indices_length
+        min_input_tokens = -(-min_extend_tokens // self.page_size) * self.page_size
```

Comment:
I am not sure I followed here, the host_indices_length also accounts for memory consumption of the request, as loading the data from host also requires allocating token slots.